include/rapidjson/reader.h

```diff
@@ -1483,14 +1484,90 @@ class GenericReader {
         // Parse minus
         bool minus = Consume(s, '-');
 
-        // Parse int: zero / ( digit1-9 *DIGIT )
+        // Parse int: zero / hexadecimal / ( digit1-9 *DIGIT )
         unsigned i = 0;
         uint64_t i64 = 0;
         bool use64bit = false;
+        bool hex = false;
```

Comment:
Ideally, not add variables for this optional feature.
I think after successfully parse a hex value, you can `goto` calling the handler.